x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/Version.java

```diff
@@ -92,7 +92,8 @@ public static Version fromString(String version) {
         String ver = "Unknown";
         String hash = ver;
 
-        if (urlStr.endsWith(".jar")) {
+        //to get spring-boot jar file name
```

Comment:
Small note: the comment suggests that the entire `if` condition is required to work with Spring Boot.
So it might be worth rephrasing to narrow it down to something like "whole JAR reference URL provided by Spring Boot" or similar.
Thanks.

Reply:
I would take out Spring Boot name from the source code altogether. This is a scenario that can be generalized, since there could be other applications or other setups that will bundle the jdbc driver jar in another jar and then discover the version this way.

Reply:
One idea for the test is to create on the fly a jar (using the Jar API) then apply the method on it.
Likely this means breaking the method a bit so the reading part is generic enough that is testable (even on hardcoded data).